homeassistant/components/nextdns/sensor.py

```diff
@@ -0,0 +1,299 @@
+"""Support for the NextDNS service."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import cast
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import NextDnsUpdateCoordinator
+from .const import (
+    ATTR_DNSSEC,
+    ATTR_ENCRYPTION,
+    ATTR_IP_VERSIONS,
+    ATTR_PROTOCOLS,
+    ATTR_STATUS,
+    DOMAIN,
+)
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass
+class NextDnsSensorRequiredKeysMixin:
+    """Class for NextDNS entity required keys."""
+
+    coordinator_type: str
+
+
+@dataclass
+class NextDnsSensorEntityDescription(
+    SensorEntityDescription, NextDnsSensorRequiredKeysMixin
+):
+    """NextDNS sensor entity description."""
+
+
+SENSORS = (
+    NextDnsSensorEntityDescription(
+        key="all_queries",
+        coordinator_type=ATTR_STATUS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:dns",
+        name="{profile_name} DNS Queries",
+        native_unit_of_measurement="queries",
+        state_class=SensorStateClass.MEASUREMENT,
```

Comment:
Since this is a # of queries, i'm wondering if this is really `TOTAL` or `TOTAL_INCREASING` since it's the total count of queries, rather than a measurement of something like "queries per second"

Caveat: I am not a sensor expert.

Reply:
`TOTAL_INCREASING` does't fit here as the statistics can be reset and the value will drop to zero.

According to the documentation `TOTAL` state class "should not be used for sensors where the absolute value is interesting instead of the accumulated growth or decline". I think the absolute value is interesting here.

Reply:
Ah, OK, I had in mind that only the rate (QPS) is interesting, but i'm used to measuring queries in another context. Thanks for confirming.